lib/postgrex.ex

```diff
@@ -491,7 +491,7 @@ defmodule Postgrex do
   @doc """
   Returns a supervisor child specification for a DBConnection pool.
   """
-  @spec child_spec([start_option]) :: Supervisor.Spec.spec()
+  @spec child_spec([start_option]) :: :supervisor.child_spec()
```

Comment:
@josevalim should we use `Supervisor.child_spec()` here instead? If so, we should do that in DBConnection too. No need to change MyXQL as it already uses that.

It's probably fine either way but I'm happy to make the change here and in other places if that'd useful.